src/storage-operators/src/oneshot_source/parquet.rs

```diff
@@ -202,15 +208,21 @@ impl<S: OneshotSource> MetadataFetch for ParquetReaderAdapter<S> {
 impl<S: OneshotSource> AsyncFileReader for ParquetReaderAdapter<S> {
     fn get_bytes(
         &mut self,
-        range: std::ops::Range<usize>,
+        range: std::ops::Range<u64>,
     ) -> BoxFuture<'_, parquet::errors::Result<Bytes>> {
         MetadataFetch::fetch(self, range)
     }
 
-    fn get_metadata(&mut self) -> BoxFuture<'_, parquet::errors::Result<Arc<ParquetMetaData>>> {
+    fn get_metadata<'a>(
+        &'a mut self,
+        _options: Option<&'a parquet::arrow::arrow_reader::ArrowReaderOptions>,
+    ) -> BoxFuture<'a, parquet::errors::Result<Arc<ParquetMetaData>>> {
         Box::pin(async move {
             let mut reader = ParquetMetaDataReader::new();
             let object_size = self.object.size();
+            let object_size = object_size.try_into().map_err(|_| {
+                ParquetError::General("Object size is too large to fit in a u64".to_string())
+            })?;
```

Comment:
We have safe wrappers for infallible conversions between `u64` and `usize`. Add a `use mz_ore::CastFrom` and then you can do `u64::cast_from(some_usize)` and `usize::cast_from(some_u64)`